docker/testing/ubuntu2004_j11.docker

```diff
@@ -172,13 +173,13 @@ RUN sh -c '\
   curl -L --fail --silent --retry 2 --retry-delay 5 --max-time 30 https://go.dev/dl/$GO_TAR -o $GO_TAR ;\
   GO_SHA="$(sha256sum $GO_TAR | cut -d" " -f2)" ;\
   echo "$GO_VERSION_SHAS" | sed "s/ /\n/g" | grep -q "$GO_SHA" || { echo "SHA256 mismatch for $GO_TAR $GO_SHA"; exit 1; } ;\
-  mkdir -p /home/cassandra/go-1.24.3; \
-  tar -C /home/cassandra/go-1.24.3 -xzf $GO_TAR ;\
+  mkdir -p $HOME/go-${GO_VERSION}; \
+  tar -C $HOME/go-${GO_VERSION} -xzf $GO_TAR ;\
   rm $GO_TAR'
 
-ENV GOROOT="/home/cassandra/go-1.24.3/go"
-ENV GOPATH="/home/cassandra/go"
-ENV PATH="$PATH:/home/cassandra/go-1.24.3/go/bin"
+ENV GOROOT="$HOME/go-${GO_VERSION}/go"
+ENV GOPATH="$HOME/go"
+ENV PATH="$PATH:$HOME/go-${GO_VERSION}/go/bin"
```

Comment:
what was wrong with the approach taken here:
https://github.com/apache/cassandra/blob/trunk/.build/docker/bullseye-build.docker#L71-L73
?